src/client/java/net/neoforged/neoforge/client/resources/NeoSplashHooks.java

```diff
@@ -0,0 +1,110 @@
+
+public class NeoSplashHooks {
+    private static final String SPLASHES_LOCATION = "texts/splashes.json";
+    private static final Map<MonthDay, List<String>> SPECIAL_SPLASHES = Maps.newHashMap();
+
+    public static List<String> prepare(List<String> vanilla, ResourceManager resourceManager, ProfilerFiller profilerFiller) {
+        SPECIAL_SPLASHES.clear();
+
+        // Vanilla splashes, including the special case "<capitalized username> IS YOU"
+        List<String> list = Lists.newArrayList(/*vanilla*/);
+        list.add("%u IS YOU");
+
+        // Special vanilla splashes
+        SPECIAL_SPLASHES.computeIfAbsent(MonthDay.of(Month.DECEMBER, 24), k -> new ArrayList<>()).add("Merry X-mas!");
+        SPECIAL_SPLASHES.computeIfAbsent(MonthDay.of(Month.JANUARY, 1), k -> new ArrayList<>()).add("Happy new year!");
+        SPECIAL_SPLASHES.computeIfAbsent(MonthDay.of(Month.OCTOBER, 31), k -> new ArrayList<>()).add("OOoooOOOoooo! Spooky!");
+
+        for (String namespace : resourceManager.getNamespaces()) {
+            try {
+                for (Resource resource : resourceManager.getResourceStack(ResourceLocation.fromNamespaceAndPath(namespace, SPLASHES_LOCATION))) {
+                    profilerFiller.push(resource.sourcePackId());
+
+                    try (Reader reader = resource.openAsReader()) {
+                        profilerFiller.push("parse");
+
+                        Splashes splashes = Splashes.CODEC.parse(JsonOps.INSTANCE, GsonHelper.parse(reader)).getOrThrow();
+                        list.addAll(splashes.splashes);
+                        for (SpecialSplashes special : splashes.specialSplashes) {
+                            for (MonthDay monthDay : special.when) {
+                                SPECIAL_SPLASHES.computeIfAbsent(monthDay, k -> new ArrayList<>()).addAll(special.splashes);
+                            }
+                        }
+
+                        profilerFiller.pop();
+                    } catch (RuntimeException runtimeexception) {
+                        LogUtils.getLogger().warn("Invalid {} in resourcepack: '{}'", "sounds.json", resource.sourcePackId(), runtimeexception);
+                    }
+
+                    profilerFiller.pop();
+                }
+            } catch (IOException ioexception) {
+                return Collections.emptyList();
+            }
+        }
+
+        return list;
+    }
+
+    public static @Nullable List<String> getSpecialSplashes() {
+        return SPECIAL_SPLASHES.get(MonthDay.now());
+    }
+
+    private static final DateTimeFormatter MONTH_DAY_PARSER = new DateTimeFormatterBuilder()
+            .appendValue(MONTH_OF_YEAR, 2)
+            .appendLiteral('-')
+            .appendValue(DAY_OF_MONTH, 2)
+            .toFormatter();
+
+    public static final Codec<MonthDay> MONTH_DAY_CODEC = Codec.withAlternative(
+            Codec.STRING.xmap(s -> MONTH_DAY_PARSER.parse(s, MonthDay::from), MONTH_DAY_PARSER::format),
```

Comment:
It might be useful to use `comapFlatMap` so that if the formatter throws an exception, it could be better wrapped into a `DataResult` for reporting